File: 2022/22/javascript/main.mjs
import fs from "node:fs";
import Map from "./map.mjs";
import { moveGrid, moveCube } from "./lib.mjs";

// --- Constants ---
const INPUT_FILE = "../input/input.txt";

// --- Read the input file ---
const text = fs.readFileSync(INPUT_FILE, "utf8");
const i = text.indexOf("\n\n");
const data = text.substring(0, i);
const code = text.substring(i + 2);

// --- Puzzle 1 ---
let map = new Map(data);
map.path(code, moveGrid);
let password = map.getPassword();
console.log("1. Password:", password.toLocaleString());

// --- Puzzle 2 ---
map = new Map(data);
map.path(code, moveCube);
password = map.getPassword();
console.log("2. Password:", password.toLocaleString());

File: 2022/22/python/main.py
from lib import Layout, move_grid, move_cube

# --- Constants ---
INPUT_FILE = "../input/input.txt"

# --- Read the input file ---
with open(INPUT_FILE, "r") as file:
    text = file.read().rstrip()
    i = text.index("\n\n")
    grid = text[0:i]
    path = text[i + 2:]

# --- Puzzle 1 ---
layout = Layout(grid)
layout.path(path, move_grid)
password = layout.get_password()
print(f"1. Password: {password}")

# --- Puzzle 2 ---
layout = Layout(grid)
layout.path(path, move_cube)
password = layout.get_password()
print(f"2. Password: {password}")
